jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java

```diff
@@ -253,13 +396,24 @@ private static FileEntriesLayer addDirectoryContentsToLayer(
     FileEntriesLayer.Builder builder = FileEntriesLayer.builder().setName(layerName);
     new DirectoryWalker(sourceRoot)
         .filterRoot()
-        .filter(path -> Files.isDirectory(path) || pathFilter.test(path))
+        .filter(path -> pathFilter.test(path))
         .walk(
             path -> {
-              AbsoluteUnixPath pathOnContainer =
-                  basePathInContainer.resolve(sourceRoot.relativize(path));
-              builder.addEntry(path, pathOnContainer);
+              builder.addEntry(path, basePathInContainer.resolve(sourceRoot.relativize(path)));
             });
     return builder.build();
   }
+
+  private static Predicate<Path> isInListedDirectoryOrIsSameFile(
+      List<String> layerContents, Path localExplodedJarRoot) {
+    Predicate<Path> predicate = Predicates.alwaysFalse();
```

Comment:
Start this with checking a file and remove the AND operation at the caller site?
```java
Predicate<Path> predicate = Files::isRegularFile;
```

Reply:
hm this method needs to check if the path is in **one** of the directories listed or if the path is the same as **one** of the files listed, which requires us to use the `or` condition. Since checking for a file requires `and` (we always want to check or this), starting with this predicate is a little challenging. I added it at the end of the method, once all the predicates were checked, but please let me know if I'm missing something here.

Reply:
Ah, you're right. Good catch.